### PR TITLE
Fix segfault during storage size metric collection

### PR DESCRIPTION
**Pull request by dgrisonnet**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:

Fix a segfault when collecting the storage size metrics when the getters used to collect the data on etcd haven't been initialized properly. This happens when the EtcdOptions are not applied which is the case for aggregated apiservers that don't care about storage.

From a downstream CI run:

```go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x20351f5]

goroutine 301 [running]:
k8s.io/apiserver/pkg/storage/etcd3/metrics.(*monitorCollector).CollectWithStability(0x99ae0a?, 0x0?)
	k8s.io/apiserver@v0.28.0-rc.0/pkg/storage/etcd3/metrics/metrics.go:271 +0x35
k8s.io/component-base/metrics.(*BaseStableCollector).Collect.func1()
	k8s.io/component-base@v0.28.0-rc.0/metrics/collector.go:83 +0x2f
created by k8s.io/component-base/metrics.(*BaseStableCollector).Collect
	k8s.io/component-base@v0.28.0-rc.0/metrics/collector.go:82 +0x8d
```

#### Does this PR introduce a user-facing change?

```release-note
k8s.io/apiserver: Fixed a nil panic attempting to collect etcd metrics in aggregated API servers that don't initialize etcd storage
```

### CODE REVIEW

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Comment by dinhxuanvu** [comment]
@logicalhan @deads2k Is it too late to tag milestone 1.28 for this PR?

**Comment by logicalhan** [comment]
> @logicalhan @deads2k Is it too late to tag milestone 1.28 for this PR?

This looks like a bugfix, we should probably take it.

**Comment by dinhxuanvu** [comment]
David is out on vacation. @liggitt Would you mind tagging the milestone 1.28 for this PR? Thanks.

**Comment by liggitt** [comment]
/kind regression
/milestone v1.28
/hold for coordination with release folks
(this seems important for 1.28.0, but small enough that it could go in after rc1 without needing another rc)

**Comment by liggitt** [comment]
is this a regression in 1.28 relative to 1.27?

**Comment by dinhxuanvu** [comment]
> is this a regression in 1.28 relative to 1.27?

It is not given the code is added in 1.28 I believe.

**Comment by liggitt** [comment]
rc1 cut is in progress, once that completes, will unhold this and merge to be available in 1.28.0

**Pull request merged**
